lib/engine/graph.rb

```diff
@@ -20,9 +20,10 @@ def clear
       @connected_paths.clear
       @reachable_hexes.clear
       @tokens.clear
+      @routes.clear
```

Comment:
this is not ideal. that will make calculation expensive for all games

Reply:
Without it, it wouldn't recalculate properly when you were in the optional state.

Most of the time it'd be recalculating it anyway for selecting routes.